hazelcast/include/hazelcast/client/impl/HazelcastClientInstanceImpl.h

```diff
@@ -420,15 +420,15 @@ namespace hazelcast {
                 std::auto_ptr<internal::nearcache::NearCacheManager> nearCacheManager;
                 spi::impl::ClientClusterServiceImpl clusterService;
                 boost::shared_ptr<spi::impl::ClientPartitionServiceImpl> partitionService;
-                std::auto_ptr<spi::impl::ClientExecutionServiceImpl> executionService;
+                boost::shared_ptr<spi::impl::ClientExecutionServiceImpl> executionService;
```

Comment:
Yes, it works since the shared_ptr constructor which accepts the auto_ptr a argument takes ownership of the auto_ptr. I will change the method signature to make this more visible.